src/redux/addCash.ts

```diff
@@ -35,13 +49,118 @@ const ADD_CASH_ORDER_FAILURE = 'addCash/ADD_CASH_ORDER_FAILURE';
 const ADD_CASH_CLEAR_STATE = 'addCash/ADD_CASH_CLEAR_STATE';
 
 // -- Actions ---------------------------------------- //
-let orderStatusHandle = null;
-let transferHashHandle = null;
+
+/**
+ * Represents the state of the `addCash` reducer.
+ */
+interface AddCashState {
+  /**
+   * The status of the current Wyre order.
+   */
+  currentOrderStatus: null | WyreOrderStatusType;
+
+  /**
+   * The current Wyre transfer ID.
+   */
+  currentTransferId: null | string;
+
+  /**
+   * The current Wyre transfer or order error, if there is one.
+   */
+  error: {} | WyreError;
+
+  /**
+   * An array of past transactions.
+   */
+  purchaseTransactions: RainbowTransaction[];
+}
+
+/**
+ * An action for the `addCash` reducer.
+ */
+type AddCashAction =
+  | AddCashUpdatePurchaseTransactionsAction
+  | AddCashClearStateAction
+  | AddCashOrderFailureAction
+  | AddCashUpdateCurrentOrderStatus
+  | AddCashUpdateCurrentTransferId
+  | AddCashResetCurrentOrderAction
+  | AddCashOrderCreationFailureAction;
+
+/**
+ * The action for updating the `purchaseTransactions` field.
+ */
+interface AddCashUpdatePurchaseTransactionsAction {
+  type: typeof ADD_CASH_UPDATE_PURCHASE_TRANSACTIONS;
+  payload: RainbowTransaction[];
+}
+
+/**
+ * The action for clearing the `addCash` state.
+ */
+interface AddCashClearStateAction {
+  type: typeof ADD_CASH_CLEAR_STATE;
+}
+
+/**
+ * The action used when a Wyre error is received.
+ */
+interface AddCashOrderFailureAction {
+  type: typeof ADD_CASH_ORDER_FAILURE;
+  payload: WyreError;
+}
+
+/**
+ * The action for updating the Wyre order status.
+ */
+interface AddCashUpdateCurrentOrderStatus {
+  type: typeof ADD_CASH_UPDATE_CURRENT_ORDER_STATUS;
+  payload: WyreOrderStatusType;
+}
+
+/**
+ * The action for updating the Wyre transfer ID.
+ */
+interface AddCashUpdateCurrentTransferId {
+  type: typeof ADD_CASH_UPDATE_CURRENT_TRANSFER_ID;
+  payload: string;
+}
+
+/**
+ * The action for resetting the current Wyre order.
+ */
+interface AddCashResetCurrentOrderAction {
+  type: typeof ADD_CASH_RESET_CURRENT_ORDER;
+}
+
+/**
+ * The action used when Wyre order creation fails.
+ */
+interface AddCashOrderCreationFailureAction {
+  type: typeof ADD_CASH_ORDER_CREATION_FAILURE;
+  payload: WyreError;
+}
+
+/**
+ * The ID for a timeout that fetches the order status, or null.
+ */
+let orderStatusHandle: null | ReturnType<typeof setTimeout> = null;
+
+/**
+ * The ID for a timeout that fetches a transfer hash, or null.
+ */
+let transferHashHandle: null | ReturnType<typeof setTimeout> = null;
 
 const MAX_TRIES = 10 * 60;
 const MAX_ERROR_TRIES = 3;
 
-export const addCashLoadState = () => async (dispatch, getState) => {
+/**
+ * Loads past purchase transactions into the reducer's state from local storage.
+ */
+export const addCashLoadState = () => async (
+  dispatch: Dispatch<AddCashUpdatePurchaseTransactionsAction>,
```

Comment:
The remaining reducers all follow the pattern of using `Dispatch<...>` instead of `AppDispatch` due to the issue from earlier where `AppDispatch` doesn't do type checking (yet).